presto-base-arrow-flight/src/test/java/com/facebook/plugin/arrow/TestArrowFlightNativeQueries.java

```diff
@@ -0,0 +1,427 @@
+                .useTls(certChainFile, privateKeyFile)
+                .build();
+
+        server.start();
+        log.info("Server listening on port %s", server.getPort());
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void close()
+            throws InterruptedException
+    {
+        arrowFlightQueryRunner.close();
+        server.close();
+        allocator.close();
+    }
+
+    public static int getServerPort()
+    {
+        return 9443;
+    }
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        Path prestoServerPath = Paths.get(getProperty("PRESTO_SERVER")
+                        .orElse("_build/debug/presto_cpp/main/presto_server"))
+                .toAbsolutePath();
+        assertTrue(Files.exists(prestoServerPath), format("Native worker binary at %s not found. Add -DPRESTO_SERVER=<path/to/presto_server> to your JVM arguments.", prestoServerPath));
+        log.info("Using PRESTO_SERVER binary at %s", prestoServerPath);
+
+        ImmutableMap<String, String> coordinatorProperties = ImmutableMap.of("native-execution-enabled", "true");
+        String flightCertPath = Paths.get("src/test/resources/server.crt").toAbsolutePath().toString();
+
+        return ArrowFlightQueryRunner.createQueryRunner(getServerPort(), getNativeWorkerSystemProperties(), coordinatorProperties, getExternalWorkerLauncher(prestoServerPath.toString(), getServerPort(), flightCertPath));
+    }
+
+    @Override
+    protected FeaturesConfig createFeaturesConfig()
+    {
+        return new FeaturesConfig().setNativeExecutionEnabled(true);
+    }
+
+    @Test
+    public void testBasicSelect()
+    {
+        assertQuery("SELECT * FROM nation");
+    }
+
+    @Test
+    public void testFiltersAndProjections1()
+    {
+        assertQuery("SELECT * FROM nation");
+        assertQuery("SELECT * FROM nation WHERE nationkey = 4");
+        assertQuery("SELECT * FROM nation WHERE nationkey <> 4");
+        assertQuery("SELECT * FROM nation WHERE nationkey < 4");
+        assertQuery("SELECT * FROM nation WHERE nationkey <= 4");
+        assertQuery("SELECT * FROM nation WHERE nationkey > 4");
+        assertQuery("SELECT * FROM nation WHERE nationkey >= 4");
+        assertQuery("SELECT * FROM nation WHERE nationkey BETWEEN 3 AND 7");
+        assertQuery("SELECT * FROM nation WHERE nationkey IN (1, 3, 5)");
+        assertQuery("SELECT * FROM nation WHERE nationkey NOT IN (1, 3, 5)");
+        assertQuery("SELECT * FROM nation WHERE nationkey NOT IN (1, 8, 11)");
+        assertQuery("SELECT * FROM nation WHERE nationkey NOT IN (1, 2, 3)");
+        assertQuery("SELECT * FROM nation WHERE nationkey NOT IN (-14, 2)");
+        assertQuery("SELECT * FROM nation WHERE nationkey NOT IN (1, 2, 3, 4, 5, 10, 11, 12, 13)");
+        // Java coordinator/workers causes these queries to fail, even though the INT_MAX ones work on cpp
```

Comment:
Can you elaborate this?

Reply:
I copied some tests from AbstractTestNativeGeneralQueries, I left the test as it was, including this comment, which wasn't from me https://github.ibm.com/lakehouse/presto/blob/719bfe2a4d35703e04b3127f6db9d37b94e8dc8c/presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeGeneralQueries.java#L161

I'm not exactly sure why they fail for Java either, I can remove the commented out tests though, since it is a copy